compiler/front/options.nim

```diff
@@ -432,13 +254,103 @@ type
       debugUtilsStack*: seq[string] ## which proc name to stop trace output
       ## len is also used for output indent level
 
+template passField(fieldname, fieldtype: untyped): untyped =
+  proc `fieldname`*(conf: ConfigRef): fieldtype =
+    conf.active.fieldname
+
+  proc `fieldname=`*(conf: ConfigRef, val: fieldtype) =
+      conf.active.fieldname = val
+
+passField(backend, TBackend)
+passField(target, Target)
+
+template passSetField(fieldname, fieldtype, itemtype: untyped): untyped =
```

Comment:
Can we collect all these helper templates together that way you just scroll past them and read the defines all in one continuous chunk rather than being interrupted.